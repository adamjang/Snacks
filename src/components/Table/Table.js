import React               from 'react'
import PropTypes           from 'prop-types'
import Radium              from 'radium'
import { spacing, colors } from '../../styles'
import responsive          from '../../styles/responsive'

const styles = {
  table: {
    borderCollapse: 'collapse',
    borderRadius: '4px',
    borderSpacing: 0,
    boxShadow: '0 1px 2px #AAA',
    overflow: 'hidden',
    width: '100%',
  },

  cell: {
    background: colors.WHITE,
    fontSize: '14px',
    fontWeight: 400,
    height: spacing.XL,
    lineHeight: `${spacing.XL}px`,
    padding: `0 ${spacing.SM}px`,
    textAlign: 'left',
    whiteSpace: 'nowrap',

    [responsive.md]: {
      height: spacing.LG,
      lineHeight: `${spacing.LG}px`,
      padding: `0 ${spacing.XS}px`,
    },

    [responsive.sm]: {
      fontSize: '12px',
      height: spacing.MD,
      lineHeight: `${spacing.MD}px`,
    }
  },

  cellAlt: {
    background: colors.GRAY_97,
  }
}
styles.cell[responsive.xs] = styles.cell[responsive.sm]

const Table = props => {
  const { withHeader, definition, data } = props

  return (
    <table style={styles.table}>
      {!withHeader ? null : (
        <thead>
          <tr>
            {definition.map((def, index) => (
              <th key={index} style={styles.cell}>
                {def.header !== undefined ? def.header : def.attribute}
              </th>
            ))}
          </tr>
        </thead>
      )}

      <tbody>
        {data.map((row, index) => (
          <tr key={index}>
            {definition.map((def, cellIndex) => (
              <td
                key={cellIndex}
                style={[
                  styles.cell,
                  index % 2 === 0 && styles.cellAlt
                ]}
                {...Object.assign({}, def, { header: null, attribute: null, cellRender: null })}
              >
                {def.cellRender ? def.cellRender(row[def.attribute], index, row) : row[def.attribute]}
              </td>
            ))}
          </tr>
        ))}
      </tbody>
    </table>
  )
}

Table.propTypes = {
  data: PropTypes
    .arrayOf(PropTypes.object)
    .isRequired,

  definition: (props, propName) => {
    const definition = props[propName]

    if (!Array.isArray(definition)) {
      throw new Error('Table required definition prop must be an Array')
    }

    for (const cell of definition) {
      if (cell.attribute === undefined) {
        throw new Error('All Table definition object must have an .attribute')
      }
    }
  },

  withHeader: PropTypes.bool,
}

Table.defaultProps = {
  withHeader: true,
}

export default Radium(Table)
