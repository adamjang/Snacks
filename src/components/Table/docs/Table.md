Table example:

```jsx
    <div>
      <Table
        data={[
          { name: 'Apple', title: 'fruit' },
          { name: 'Carrot', title: 'vegetable' },
          { name: 'Lemon', title: 'fruit' },
          { name: 'Avocado', title: 'VEGETABLE' }
        ]}
        definition={[
          { header: 'Name', attribute: 'name' },
          { header: 'Title', attribute: 'title', cellRender: (name, rowIndex, rowData) => (<span>{name.toLowerCase()}</span>) }
        ]}
      />
    </div>
```

You can apply row click events with the prop `onRowClick`

Individual cells can have their own click handler as well, using `onClick` within a `definition`. These will override any applied `onRowClick`.

```jsx
    <div>
      <Table
        withHeader={false}
        data={[
          { name: 'Apple', title: 'fruit', emoji: '🍎' },
          { name: 'Carrot', title: 'vegetable', emoji: '🥕' },
          { name: 'Lemon', title: 'fruit', emoji: '🍋' },
          { name: 'Avocado', title: 'VEGETABLE', emoji: '🥑' }
        ]}
        definition={[
          { header: 'Name', attribute: 'name', onClick: rowData => alert(`Cell click for "${rowData.name}"`) },
          { header: 'Title', attribute: 'title', cellRender: (name, rowIndex, rowData) => (<span>{name.toLowerCase()}</span>) }
        ]}
        onRowClick={rowData => {
          alert(rowData.emoji.repeat(3))
        }}
      />
    </div>
```
