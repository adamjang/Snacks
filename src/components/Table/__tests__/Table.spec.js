import React         from 'react'
import renderer      from 'react-test-renderer'
import { StyleRoot } from 'radium'
import { mount }     from 'enzyme'
import { spy }       from 'sinon'
import Table         from '../Table'

it('renders Table with minimal definition correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <Table
          data={[
            { name: 'Apple', type: 'fruit' },
            { name: 'Carrot', type: 'vegetable' },
            { name: 'Lemon', type: 'fruit' },
            { name: 'Avocado', type: 'vegetable' }
          ]}
          definition={[
            { attribute: 'name' },
            { attribute: 'type' }
          ]}
        />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Table with definition headers correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <Table
          data={[
            { name: 'Apple', type: 'fruit', qty: 3 },
            { name: 'Carrot', type: 'vegetable', qty: 99 },
            { name: 'Lemon', type: 'fruit', qty: 12 },
            { name: 'Avocado', type: 'vegetable', qty: 0 }
          ]}
          definition={[
            { header: 'Type', attribute: 'type' },
            { header: 'Name', attribute: 'name' },
            { header: 'Quantity', attribute: 'qty' }
          ]}
        />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Table with cellRender methods correctly', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <Table
          data={[
            { name: 'Apple', type: 'fruit', qty: 3 },
            { name: 'Carrot', type: 'vegetable', qty: 99 },
            { name: 'Lemon', type: 'fruit', qty: 12 },
            { name: 'Avocado', type: 'vegetable', qty: 0 }
          ]}
          definition={[
            { header: 'Name', attribute: 'name', cellRender: name => name.toLowerCase() },
            { header: 'Type', attribute: 'type', cellRender: type => type.charAt(0) },
            { header: 'Quantity', attribute: 'qty', cellRender: (qty, index, row) => `[${index}]: ${qty} ${row.name}s left` }
          ]}
        />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders Table without header if specified', () => {
  const tree = renderer.create(
    <StyleRoot>
      <div>
        <Table
          data={[
            { name: 'Apple', type: 'fruit' },
            { name: 'Carrot', type: 'vegetable' },
            { name: 'Lemon', type: 'fruit' },
            { name: 'Avocado', type: 'vegetable' }
          ]}
          definition={[
            { attribute: 'name' },
            { attribute: 'type' }
          ]}
          withHeader={false}
        />
      </div>
    </StyleRoot>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
