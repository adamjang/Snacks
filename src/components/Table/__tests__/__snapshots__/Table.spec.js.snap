// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Table with cellRender methods correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <table
      data-radium={true}
      style={
        Object {
          "borderCollapse": "collapse",
          "borderRadius": "4px",
          "borderSpacing": 0,
          "boxShadow": "0 1px 2px #AAA",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <thead
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Name
          </th>
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Type
          </th>
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Quantity
          </th>
        </tr>
      </thead>
      <tbody
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            apple
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            f
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            [0]: 3 Apples left
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            carrot
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            v
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            [1]: 99 Carrots left
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            lemon
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            f
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            [2]: 12 Lemons left
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            avocado
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            v
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            [3]: 0 Avocados left
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders Table with definition headers correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <table
      data-radium={true}
      style={
        Object {
          "borderCollapse": "collapse",
          "borderRadius": "4px",
          "borderSpacing": 0,
          "boxShadow": "0 1px 2px #AAA",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <thead
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Type
          </th>
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Name
          </th>
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Quantity
          </th>
        </tr>
      </thead>
      <tbody
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Apple
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            3
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Carrot
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            99
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Lemon
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            12
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Avocado
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            0
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders Table with minimal definition correctly 1`] = `
<div
  data-radium={true}
>
  <div>
    <table
      data-radium={true}
      style={
        Object {
          "borderCollapse": "collapse",
          "borderRadius": "4px",
          "borderSpacing": 0,
          "boxShadow": "0 1px 2px #AAA",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <thead
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            name
          </th>
          <th
            data-radium={true}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            type
          </th>
        </tr>
      </thead>
      <tbody
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Apple
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Carrot
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Lemon
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Avocado
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;

exports[`renders Table without header if specified 1`] = `
<div
  data-radium={true}
>
  <div>
    <table
      data-radium={true}
      style={
        Object {
          "borderCollapse": "collapse",
          "borderRadius": "4px",
          "borderSpacing": 0,
          "boxShadow": "0 1px 2px #AAA",
          "overflow": "hidden",
          "width": "100%",
        }
      }
    >
      <tbody
        data-radium={true}
      >
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Apple
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Carrot
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Lemon
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#F7F7F7",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            fruit
          </td>
        </tr>
        <tr
          data-radium={true}
        >
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            Avocado
          </td>
          <td
            attribute={null}
            cellRender={null}
            data-radium={true}
            header={null}
            style={
              Object {
                "background": "#FFFFFF",
                "fontSize": "14px",
                "fontWeight": 400,
                "height": 48,
                "lineHeight": "48px",
                "padding": "0 16px",
                "textAlign": "left",
                "whiteSpace": "nowrap",
              }
            }
          >
            vegetable
          </td>
        </tr>
      </tbody>
    </table>
  </div>
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": "",
      }
    }
  />
</div>
`;
